apps/web/src/pages/templates/editor/Preview.tsx

```diff
@@ -28,29 +32,41 @@ export const Preview = ({ activeStep, view }: { activeStep: number; view: string
     control,
   });
 
+  const variables = useWatch({
+    name: `steps.${activeStep}.template.variables`,
+    control,
+  });
+
   const { integrations = [] } = useIntegrations();
   const [integration, setIntegration]: any = useState(null);
   const [content, setContent] = useState<string>('<html><head></head><body><div></div></body></html>');
   const { isLoading, mutateAsync } = useMutation(previewEmail);
+  const processedVariables = useProcessVariables(variables);
+  const [payloadValue, setPayloadValue] = useState('{}');
 
   useEffect(() => {
-    if (contentType !== 'editor') {
-      if (!htmlContent) {
-        return;
-      }
-      setContent(htmlContent);
+    setPayloadValue(processedVariables);
+  }, [processedVariables, setPayloadValue]);
 
-      return;
-    }
-    mutateAsync({
-      content: editorContent,
-      contentType,
-    }).then((result: { html: string }) => {
+  const parseContent = (args: {
+    contentType: MessageTemplateContentType;
+    content: string | IEmailBlock[];
+    payload: any;
+  }) => {
+    mutateAsync(args).then((result: { html: string }) => {
       setContent(result.html);
 
       return result;
     });
-  }, [contentType, htmlContent, editorContent, setContent, mutateAsync]);
+  };
+
+  useEffect(() => {
+    parseContent({
+      contentType,
+      content: contentType === 'editor' ? editorContent : htmlContent,
+      payload: processedVariables,
+    });
+  }, [contentType, htmlContent, editorContent, setContent, mutateAsync, processedVariables]);
```

Comment:
I think `setContent` and `mutateAsync` are not needed dependencies, but `parseContent` would be.